includes/TemplateParser.php

```diff
@@ -36,34 +40,65 @@ function commonsbooking_parse_shortcode( $tag ) {
  */
 function commonsbooking_parse_template_callback( $match, array $objects = [] ) {
 
-	if ( isset( $match[0] ) ) {
-		$match = $match[0];
-		$match = preg_replace( '/(\{\{)|(\}\})/m', '', $match );
-		// we accept : and # as separator cause the : delimiter wasn't working when using the template tag in a href links in the template (like <a href="{{xxx#yyyy}}"></a>)
-		$path = preg_split( '/(\:|\#)/', $match, 2 );
-		if ( isset( $path[0] ) and isset( $path[1] ) ) {
-			$post = null;
-			if ( array_key_exists( $path[0], $objects ) ) {
-				$post = $objects[ $path[0] ];
-			}
-
-			return CB::get( commonsbooking_getCBType($path[0]), $path[1], $post );
-		}
-	}
+    if ( isset( $match[0] ) ) {
+        $match = $match[0];
+
+        // extract the html before part, looking for {{[*] pattern
+        $html_before = null;
+        if ( preg_match( '/(\{\{\[[^\]]*\])/m', $match, $html_before ) === 1 ) {
+            $html_before = commonsbooking_sanitizeHTML( preg_replace( '/(\{\{)|(\}\})|(\[)|(\])/m', '', $html_before[0] ) );
+        } else {
+            $html_before = '';
+        }
+
+        // extract the html after part looking for [*]}} pattern
+        $html_after = null;
+        if ( preg_match( '/(\[[^\]]*\]\}\})/m', $match, $html_after ) === 1 ) {
```

Comment:
Wie oben nur anders: (\[.*?\]\}\})

Reply:
@markus-mw In einem Test nach folgendem Pattern matcht deine Variante jedoch nicht zuverlässig. Der html after Teil wird nicht separat erkannt. 
siehe Hier: https://www.phpliveregex.com/p/Ee3
Meine Version erzeugt das hier: https://www.phpliveregex.com/p/Ee4
Woran liegts? (Rein aus Interesse, um das zu verstehen)